regressors/include/iDynTree/Regressors/DynamicsRegressorGenerator.h

```diff
@@ -0,0 +1,312 @@
+    std::string getDescriptionOfLinks();
+
+    /**
+     * Get name of the base link.
+     *
+     */
+    std::string getBaseLinkName();
+
+
+    //@}
+
+
+    /**
+     * @name Methods to access the underlyng model for the robot and the sensors
+     * (todo)
+     */
+    //@{
+
+    //const UndirectedTree & getRobotModel();
+    //const SensorsTree    & getSensorsModel();
+
+    //@}
+
+    /**
+      * @name Methods to submit the input data for calculating the regressor and the known terms
+      * This methods are used to submit the input data (such as the robot state and the sensor readings)
+      * to calculate the regressor and the known terms
+      */
+    //@{
+
+    /**
+     * Set the state for the robot (floating base)
+     * \todo TODO define a RobotState class.
+     */
+    bool setRobotState(const iDynTree::VectorDynSize& q,
+                       const iDynTree::VectorDynSize& q_dot,
+                       const iDynTree::VectorDynSize& q_dotdot,
+                       const iDynTree::Transform &world_T_base,
+                       const iDynTree::Twist& base_velocity,
+                       const iDynTree::Twist& base_acceleration
+                       const iDynTree::Twist& world_gravity);
+
+    /**
+     * Set the state for the robot (fixed base)
+     * Same as setRobotState, but with:
+     *  world_T_base      = iDynTree::Transform::Identity()
+     *  base_velocity     = iDynTree::Twist::Zero();
+     *  base_acceleration = iDynTree::Twist::Zero();
+     *
+     * \todo TODO define a RobotState class.
+     */
+    bool setRobotState(const iDynTree::VectorDynSize &q,
+                       const iDynTree::VectorDynSize &q_dot,
+                       const iDynTree::VectorDynSize &q_dotdot,
+                       const iDynTree::Twist& world_gravity);
+
+    /**
+     * Get a reference to the SensorsMeasurements structure, where
+     * sensors readings can be set for use of the dynamics regressor generator.
+     */
+    //SensorsMeasurements & getSensorsMeasurements();
+
+    //@}
+
+    /** @name Methods to calculate the regressor
+       */
+    //@{
+
+    /**
+     * Return the compute regressor (computed from the robot state) and the corresponding output (calculated from sensor readings)
+     * @param regressor a getNrOfOutputs() times getNrOfUnknownParameters() Matrix
+     * @param known_terms a getNrOfOutputs() parameters Vector
+     * @return true if all went well, false if there was an error
+     */
+    bool computeRegressor(iDynTree::MatrixDynSize & regressor,
+                          iDynTree::VectorDynSize & known_terms);
+
+    /**
+     * Return the dynamics parameters related to this regressor contained in the model.
+     *
+     * @param[out] values a Vector of size getNrOfParameters() ,
+     * @return true if all went well, false if there was an error
+     */
+    bool getModelParameters(iDynTree::VectorDynSize & values);
+
+    //@}
+
+private:
+
+};
+
+}
+
+}
+
+
+#endif
```

Comment:
@traversaro I've added some comments above. Moreover:

- What about the computation of the identifiable subspace via Gautier's algorithm? Is it planned to wrap it separately from the regressor generator?